protocols/luci-proto-wireguard/htdocs/luci-static/resources/protocol/wireguard.js

```diff
@@ -55,6 +62,18 @@ return network.registerProtocol('wireguard', {
 		o.validate = validateBase64;
 		o.rmempty = false;
 
+		o = s.taboption('general', form.Button, 'generate_key', _('Generate Key'), _(''));
+		o.inputstyle = 'apply';
+		o.onclick = L.bind(function(ev) {
```

Comment:
Please use the following code (and add an additional `'require ui';` to the view):

```javascript
ui.createHandlerFn(this, function(ev) {
    return generateKey().then(function(keypair) {
        var keyInput = document.getElementById("widget.cbid.network." + s.section + ".private_key"),
           changeEvent = new Event('change');

        keyInput.value = keypair.priv || '';
        keyInput.dispatchEvent(changeEvent);
    });
});
```

This will take care of disabling the button and displaying a loading indicator while the promise is resolving.

Reply:
Thanks for the tip. Changed.